### PR TITLE
Docs: Fix incorrect indents in `c-api/type.rst`

### PR DESCRIPTION
**Pull request by ghost**
Fixed some indents so the document is rendered correctly.

----
📚 Documentation preview 📚: https://cpython-previews--127449.org.readthedocs.build/

### CODE REVIEW

File: Doc/c-api/type.rst [hunk 1]
```diff
@@ -557,20 +557,20 @@ The following functions and structs are used to create
       To avoid issues, use the *bases* argument of
       :c:func:`PyType_FromSpecWithBases` instead.
 
-     .. versionchanged:: 3.9
+      .. versionchanged:: 3.9
 
-        Slots in :c:type:`PyBufferProcs` may be set in the unlimited API.
+         Slots in :c:type:`PyBufferProcs` may be set in the unlimited API.
```
Comment (picnixz) [inline Doc/c-api/type.rst:561]: There shouldn't be a newline between `.. versionchanged` and what has changed (the previous text was already wrong). See https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-versionadded.

File: Doc/c-api/type.rst [hunk 2]
```diff
@@ -557,20 +557,20 @@ The following functions and structs are used to create
...
+         :c:member:`~PyBufferProcs.bf_releasebuffer` are now available
+         under the :ref:`limited API <limited-c-api>`.
 
-     .. versionchanged:: 3.14
+      .. versionchanged:: 3.14
 
-        The field :c:member:`~PyTypeObject.tp_vectorcall` can now set
-        using ``Py_tp_vectorcall``.  See the field's documentation
-        for details.
+         The field :c:member:`~PyTypeObject.tp_vectorcall` can now set
```
Comment (picnixz) [inline Doc/c-api/type.rst:569]: Ditto.

**Review by picnixz — APPROVED** [review]

**Review by Eclips4 — APPROVED** [review]
Thank you. LGTM.

**Comment by Eclips4** [comment]
@koyuki7w could you please create a manual backports to 3.12 and 3.13?

**Comment by koyuki7w** [comment]
@Eclips4 Thanks. Opened PRs for 3.12 and 3.13 backports.

**Pull request merged**